core/src/main/java/org/apache/iceberg/PartitionsTable.java

```diff
@@ -93,16 +93,36 @@ private static StaticDataTask.Row convertPartition(Partition partition) {
     return StaticDataTask.Row.of(partition.key, partition.recordCount, partition.fileCount, partition.specId);
   }
 
-  private static Iterable<Partition> partitions(StaticTableScan scan) {
+  private static Iterable<Partition> partitions(Table table, StaticTableScan scan) {
     CloseableIterable<FileScanTask> tasks = planFiles(scan);
 
-    PartitionMap partitions = new PartitionMap(scan.table().spec().partitionType());
+    PartitionMap partitions = new PartitionMap(Partitioning.partitionType(table));
     for (FileScanTask task : tasks) {
-      partitions.get(task.file().partition()).update(task.file());
+      PartitionData original = (PartitionData) task.file().partition();
+      PartitionData normalized = normalizePartition(original, Partitioning.partitionType(table));
```

Comment:
Should we reuse the result of the Partitioning.partitionType(table) invocation at line 99 rather than calling it in every iteration?

Reply:
Yea good point